providers/dns/cloudxns/cloudxns.go

```diff
@@ -1,32 +1,45 @@
 // Package cloudxns implements a DNS provider for solving the DNS-01 challenge
-// using cloudxns DNS.
+// using CloudXNS DNS.
 package cloudxns
 
 import (
-	"bytes"
-	"crypto/md5"
-	"encoding/hex"
-	"encoding/json"
+	"errors"
 	"fmt"
 	"net/http"
-	"strconv"
 	"time"
 
 	"github.com/xenolf/lego/acme"
 	"github.com/xenolf/lego/platform/config/env"
 )
 
-const cloudXNSBaseURL = "https://www.cloudxns.net/api2/"
+// Config is used to configure the creation of the DNSProvider
+type Config struct {
+	APIKey     string
+	SecretKey  string
+	TTL        int
+	HTTPClient *http.Client
+}
+
+// NewDefaultConfig returns a default configuration for the DNSProvider
+func NewDefaultConfig() *Config {
+	client := acme.HTTPClient
+	client.Timeout = time.Second * time.Duration(env.GetOrDefaultInt("CLOUDXNS_HTTP_TIMEOUT", 30))
```

Comment:
Is this not modifying the default/package level client?

Reply:
no because I use a copy of `acme.HTTPClient`